syntax = "proto3";
package mvshape_dataset;

message Rendering {
  enum Type {
    DEPTH = 0;
    NORMAL = 1;
    RGB = 2;
    VOXELS = 3;
  }
  int32 id = 1;
  string filename = 2;
  int32 resolution = 3;
  int32 num_channels = 4;
  int32 set_size = 5;
  bool is_normalized = 6;
  Type rendering_type = 7;
  string mesh_filename = 8;
  int32 object_id = 9;
  repeated float eye = 10;
  repeated float up = 11;
  repeated float lookat = 12;
  float fov = 13;
  float scale = 14;
  int32 camera_id = 15;
  string category_name = 16;
  int32 category_id = 17;
}

enum Split {
  TEST = 0;
  VALIDATION = 1;
  VALIDATION_SUBSET = 2;
  TRAIN = 3;
  TRAIN_VALIDATION = 4;
}

enum Tag {
  NOVELVIEW = 0;
  NOVELMODEL = 1;
  NOVELCLASS = 2;
  PERSPECTIVE_INPUT = 3;
  ORTHOGRAPHIC_INPUT = 4;
  PERSPECTIVE_OUTPUT = 5;
  ORTHOGRAPHIC_OUTPUT = 6;
  VIEWER_CENTERED = 7;
  OBJECT_CENTERED = 8;
  REAL_WORLD = 9;
}

enum DatasetName {
  SHREC12 = 0;
  SHAPENETCORE = 1;
  PASCAL3D = 2;
}

// TODO: Change the names of single_depth and single_depth_raw.
message Example {
  int32 id = 1;
  repeated Tag tags = 2;
  Rendering single_depth = 3;  // Normalized
  Rendering single_rgb = 4;
  Rendering multiview_depth = 5;
  Rendering multiview_normal = 6;
  Rendering multiview_rgb = 7;
  Rendering voxels = 8;
  Rendering single_depth_raw = 9;  // same as `single_depth` except it is not normalized and in higher resolution.
}

message Examples {
  DatasetName dataset_name = 1;
  Split split_name = 2;
  repeated Example examples = 3;
  repeated Tag tags = 4;
}
